tools/thanosconvert/thanosconvert.go

```diff
@@ -0,0 +1,188 @@
+package thanosconvert
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+	"github.com/thanos-io/thanos/pkg/objstore"
+
+	"github.com/cortexproject/cortex/pkg/storage/bucket"
+	cortex_tsdb "github.com/cortexproject/cortex/pkg/storage/tsdb"
+)
+
+// ThanosBlockConverter converts blocks written by Thanos to make them readable by Cortex
+type ThanosBlockConverter struct {
+	logger log.Logger
+	bkt    objstore.Bucket
+	dryRun bool
+}
+
+type PerUserResults struct {
+	FailedBlocks, ConvertedBlocks, UnchangedBlocks []string
+}
+
+func NewPerUserResults() PerUserResults {
+	return PerUserResults{
+		FailedBlocks:    []string{},
```

Comment:
Nit: this creates 0-sized slice, which is unnecessary. As it is now, `PerUserResults{}` is perfectly fine default value, and doesn't even need "New" function -- which isn't used anywhere.

(Furthermore, since methods are defined on `*PerUserResults`, this method should return pointer. But I'd suggest to remove it completely.)

Reply:
Turns out I wasn't even using this constructor anyway 😄